### PR TITLE
Refactor `OnStakingUpdate` interface to be safer to use.

### PR DESCRIPTION
**Pull request by gpestana**

### CODE REVIEW

**Comment by gpestana** [comment]
bot fmt

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4860727

File: substrate/frame/staking/src/pallet/impls.rs [hunk 1]
```diff
@@ -997,17 +997,19 @@ impl<T: Config> Pallet<T> {
...
 				Nominators::<T>::insert(who, nominations);
-				T::EventListeners::on_nominator_add(who);
+				T::EventListeners::on_nominator_add(who, targets);
 			},
 			(_, true) => {
 				// update nominations or un-chill nominator.
 				let prev_nominations = Self::nominations(who).unwrap_or_default();
 				Nominators::<T>::insert(who, nominations);
-				T::EventListeners::on_nominator_update(who, prev_nominations);
+				T::EventListeners::on_nominator_update(who, prev_nominations, targets);
```
Comment (Ank4n) [inline substrate/frame/staking/src/pallet/impls.rs:1012]: lets call them either nominations or targets but not both?

**Review by Ank4n — APPROVED** [review]
Trait changes looks fine. Since this only wants to merge to a non-master working branch, looks good to merge.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4862349

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4862348

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4862331

**Pull request merged**